### PR TITLE
feat: add pad method to SimpleTable for string padding (#28)

### LINKED ISSUE #28: Add pad() method to SimpleTable for string padding (LPAD/RPAD)
Pad string values in specified columns to a target length using a fill character. Supports both left and right padding via DuckDB's `LPAD`/`RPAD`.

## Context

No string padding method currently exists. `trim()`, `upper()`, `lower()`, and `truncate()` cover other string operations but not padding.

## Requirements

- Add a `pad()` method on `SimpleTable`
- Supports left (`LPAD`) and right (`RPAD`) padding via a `method` option
- Works on single or multiple columns
- Pre-checks string lengths; throws if any value exceeds the target length (no silent truncation)
- Returns `Promise<void>` (in-place mutation)

### Method signature

```ts
async pad(
  columns: string | string[],
  length: number,
  options: {
    character?: string;  // default: '0'
    method?: "left" | "right";  // default: 'left'
  } = {},
): Promise<void>
```

### Query helper

```ts
// src/methods/padQuery.ts
export default function padQuery(
  table: string,
  columns: string[],
  length: number,
  options: { character?: string; method?: "left" | "right" },
): string
```

Generates `UPDATE` statements using `LPAD` or `RPAD`.

### Pre-validation query

Before applying padding, run:
```sql
SELECT COUNT(*) FROM "<table>" WHERE LENGTH("<column>") > <length>
```
If count > 0, throw an error identifying the column and the number of overflowing rows.

## Acceptance Criteria

- [ ] `padQuery.ts` created in `src/methods/` generating correct `LPAD`/`RPAD` SQL
- [ ] `pad()` method added to `SimpleTable` class with full JSDoc
- [ ] Pre-validation query throws on strings exceeding target length
- [ ] Default `character` is `0`, default `method` is `left`
- [ ] Works with `string` and `string[]` for columns
- [ ] Category set to `@category Updating Data`
- [ ] Unit tests covering: left pad, right pad, custom character, multiple columns, overflow error, default values
- [ ] `deno lint`, `deno fmt`, `deno check`, and `deno test` all pass

## Technical Notes

- Follow the existing pattern of `trim()`/`upper()`/`lower()`: query helper in `src/methods/`, method on `SimpleTable` using `queryDB` + `mergeOptions`
- Use `stringToArray()` for column normalization
- DuckDB `LPAD(string, length, padding)` and `RPAD(string, length, padding)` are the underlying functions

---
_Edit (2026-05-06): Rewrote issue with full requirements, data shapes, signatures, acceptance criteria, and technical notes based on discussion._

### PR DESCRIPTION
**Pull request by nshiab**
## Summary
  
Implements the `pad` method on `SimpleTable` as specified in issue #28, allowing users to pad string values in a column to a target length.

### New Files
- `src/methods/padQuery.ts` - SQL helper function generating `LPAD`/ `RPAD` DuckDB queries
- `test/unit/methods/pad.test.ts` - 14 unit tests covering all scenarios

### Modified Files
- `src/class/SimpleTable.ts` - Added `pad(column, length, options)` method with full JSDoc

## Method Signature

```typescript
pad(
  column: string,
  length: number,
  options?: { side?: 'start' | 'end'; char?: string }
): Promise<void>;
```

## Features
- **Left-side (start) padding** (default): `await table.pad('id', 3)` → `'1'` → `'001'`
- **Right-side (end) padding**: `await table.pad('code', 5, { side: 'end' })` → `'123'` → `'12300'`
- **Custom padding character**: `await table.pad('id', 4, { char: '-' })` → `'1'` → `'---1'`
- **Default padding char**: `'0'`
- **Default side**: `'start'`
- **Null handling**: `null` values remain `null`
- **Non-string values**: Left unchanged
- **Truncation**: Strings longer than target length are truncated to fit

## Tests (14 cases)
- Standard start/end padding
- Custom character padding
- Handling of null values
- Handling of mixed types (strings vs non-strings)
- Strings longer than target length (truncated)
- Column names with spaces
- Empty strings
- Padding to length 0
- Multi-character fill strings
- Default side and char verification
- Multiple rows with null values

## Acceptance Criteria
- ✅ `pad` method implemented with specified signature
- ✅ Padding correctly applies to strings on specified side (start or end)
- ✅ Default behavior is `side: 'start'` and `char: '0'`
- ✅ `null` values remain `null`
- ✅ Non-string values (other than `null`) are ignored/unchanged
- ✅ Unit tests added covering all acceptance criteria
- ✅ JSDoc documentation added with examples

Fixes #28

### CODE REVIEW

**Comment by nshiab** [comment]
## Review: feat: add pad method to SimpleTable for string padding (#28)

### Overall Assessment

This is a well-executed implementation of the `pad` method. The PR is clean, well-tested, and follows existing code patterns in the codebase.

### Test Coverage (14 tests - all passing)

Excellent test coverage across all scenarios:
- Default behavior: left-pad with '0' (default side + char)
- Right-padding with side: "end" and default '0'
- Custom characters for both left and right padding
- Null handling: null values preserved, non-null strings padded
- Mixed types: non-string columns unaffected
- Truncation: strings longer than target length are truncated (DuckDB LPAD/RPAD behavior)
- Edge cases: empty strings, length 0, column names with spaces, multi-character fill strings
- Default verification: explicit tests confirming default side and char values

### Code Quality

- padQuery.ts: Clean, concise SQL generation using DuckDB's LPAD/RPAD. Follows the same pattern as truncateQuery.ts and capitalizeQuery.ts.
- SimpleTable.ts: The pad() method follows the established pattern - calls queryDB() with the generated SQL and mergeOptions() for logging.
- Type safety: Proper TypeScript typing with discriminated union for side option. Default parameter {} is correctly typed.

### Documentation

- JSDoc: Comprehensive with 4 examples covering the main use cases. Clear description of behavior (null handling, non-string preservation, truncation).
- llm.md: Properly updated with full documentation section matching the JSDoc.
- JSDoc format: Nested options.side / options.char convention is consistent with existing methods in the codebase (e.g., loadData options).

### No Issues Found

- No linting errors
- No type checking errors
- All 14 tests pass
- Follows existing architectural patterns
- No security concerns (SQL quoting consistent with existing codebase)

### Recommendation

Approved - ready to merge.

File: src/class/SimpleTable.ts [hunk 1]
```diff
@@ -2785,6 +2786,65 @@ export default class SimpleTable extends Simple {
     );
   }
 
+  /**
+   * Pads the strings in the specified column to a target length.
+   *
+   * Only string values are padded. `null` values remain `null`, and non-string
+   * values (other than `null`) are left unchanged. Strings longer than the
```
Comment (nshiab) [inline src/class/SimpleTable.ts]: If the column is not strings, it should throw an error. DuckDB will probably throw it and that's okay.

File: src/class/SimpleTable.ts [hunk 2]
```diff
@@ -2785,6 +2786,65 @@ export default class SimpleTable extends Simple {
...
+   * target length are truncated to fit.
```
Comment (nshiab) [inline src/class/SimpleTable.ts]: If a string is longer than the length, an error should be thrown. It's okay if it's duckdb that throws it.

Comment (nshiab) [reply src/class/SimpleTable.ts]: We could have an option to allow truncation

File: src/class/SimpleTable.ts [hunk 3]
```diff
@@ -2785,6 +2786,87 @@ export default class SimpleTable extends Simple {
...
+   *
+   * @example
+   * ```ts
+   * // Padding with a longer fill string (DuckDB repeats/truncates as needed)
+   * await table.pad("code", 6, { side: "end", char: "ab" });
+   * // Result: '1' -> '1ababa', '23' -> '23abab'
+   * ```
+   */
+  async pad(
+    column: string,
```
Comment (nshiab) [inline src/class/SimpleTable.ts:2835]: It would be nice if column would be columns: string | string[], like other methods

File: src/class/SimpleTable.ts [hunk 4]
```diff
@@ -2785,6 +2786,87 @@ export default class SimpleTable extends Simple {
...
+
+    // Validate no string value exceeds the target length
+    const values = await this.getValues(column);
+    for (const val of values) {
+      if (val !== null && typeof val === "string" && val.length > length) {
+        throw new Error(
+          `The string "${val}" in column "${column}" has length ${val.length}, which exceeds the target length ${length}.`,
+        );
+      }
+    }
```
Comment (nshiab) [inline src/class/SimpleTable.ts:2857]: This is pretty bad performance wise. We should use DuckDB. Use this.sdb.customQuery(your_query, { returnFrom: "query"}) to check this. We can retrieve the max length of the string values and compare it with the parameter length.

**Review by nshiab — COMMENTED** [review]
Fix these
